pkg/skaffold/docker/logger/log_test.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+import (
+	"context"
+	"io/ioutil"
+	"testing"
+)
+
+func TestDockerLoggerZeroValue(t *testing.T) {
```

Comment:
How is *Logger setup s.t. it can have a zero value initialization?  Is it because the types in it struct support zero value init?
```
type Logger struct {
	out         io.Writer
	tracker     *tracker.ContainerTracker
	colorPicker output.ColorPicker
	client      docker.LocalDaemon
	muted       int32
}
```

Reply:
nah this is actually because the method itself has a pointer receiver, so if the `*Logger` was nil (because it was initialized that way, e.g. `var l *Logger`), then calling `l.Start()` wouldn't actually NPE because it has a check for `if l == nil` in the code. AFAIK this is unique to Go, if you tried to call any method on a pointer receiver it would immediately NPE in something like C++.

that said, I've changed the way we instantiate these objects in code to use `Noop` implementations now instead of `nil` (e.g. for the `Logger`, if logs are muted we use `&NoopLogger{}` instead of `nil`), so theoretically this should never happen. that said, it's always possible one of these gets missed, so I'll leave the test here anyway just as a final guardrail so NPEs never get surfaced to the user.